cookbook/getting_started/17_user_memories_and_summaries.py

```diff
@@ -87,66 +87,65 @@ def print_agent_memory(agent):
     """Print the current state of agent's memory systems"""
     console = Console()
 
-    messages = []
-    session_id = agent.session_id
-    session_run = agent.memory.runs[session_id][-1]
-    for m in session_run.messages:
-        message_dict = m.to_dict()
-        messages.append(message_dict)
-
     # Print chat history
+    messages = agent.get_messages_for_session()
     console.print(
         Panel(
             JSON(
                 json.dumps(
-                    messages,
+                    [m.model_dump(include={"role", "content"}) for m in messages],
+                    indent=4,
                 ),
-                indent=4,
             ),
-            title=f"Chat History for session_id: {session_run.session_id}",
+            title=f"Chat History for session_id: {agent.session_id}",
             expand=True,
         )
     )
 
     # Print user memories
-    for user_id in list(agent.memory.memories.keys()):
-        console.print(
-            Panel(
-                JSON(
-                    json.dumps(
-                        [
-                            user_memory.to_dict()
-                            for user_memory in agent.memory.get_user_memories(
-                                user_id=user_id
-                            )
-                        ],
-                        indent=4,
-                    ),
-                ),
-                title=f"Memories for user_id: {user_id}",
-                expand=True,
+    user_memories = agent.memory_manager.get_user_memories(user_id=agent.user_id)
+    if user_memories:
+        try:
+            memories_data = []
+            for memory in user_memories:
+                try:
+                    memories_data.append(memory.to_dict())
+                except AttributeError:
+                    # Handle serialization issues by creating a simple dict
```

Comment:
does this happen??

Reply:
Hey Dirk , so to_dict() function is trying to convert int -> isoformat string 
`Error displaying memories: 'int' object has no attribute 'isoformat'` and hence that is failing

Reply:
Ah that is a bug. @manuhortet